internal/pkg/util/util.go

```diff
@@ -30,7 +30,7 @@ func ToHeaders(headersFlag []string) map[string]string {
 			headers[strings.TrimSpace(kv[0])] = ""
 			continue
 		}
-		headers[strings.TrimSpace(kv[0])] = strings.TrimSpace(kv[1])
+		headers[strings.TrimSpace(kv[0])] = placeholders.InterpolatePlaceholders(strings.TrimSpace(kv[1]))
```

Comment:
ToHeaders is only called for HTTP worker so this could work only for http requests and not grpc requests. Secondly, doing it this way will not randomise headers within a worker. to achieve that we will have to create channel for headers just like we do for headers.